### ISSUE TITLE
Improve git documentation

### ISSUE THREAD

**Issue by jzeneto** [damionx7/Zettel-Notes-Documentation#33]
Hi, the documentation for using Git needs an improving. I have a problem and I don't even imagine how to solve it, because the documentation gives me no clues.

I've tried to use Zettel Notes with an already existing GitLab **private** repository, with no success. There are three messages:

- in Git configuration, when I press the "check mark" button at top right: "Checking connection" and then "Success"
- after configuration: "`org.eclipse.jgit.api.errors.NoHeadException`: No HEAD exists and no explicit starting revision was specified"
- in phone notification area: "Error - Login failed : Ref HEAD can not be resolved"

I've used a personal access token, with only "api" permission, together with my username, the repository address and branch ("master").

**Comment by jzeneto** [comment]
It worked, but I think there's a bug, or documentation is wrong:

It only worked when I put `HEAD` as branch. I've tried to put `master` (my repository is old, it uses `master`, not `main`) without success. But in docs there's no mention to `HEAD`, as I remember.

I understand you develop this app in your spare time, but please, write a more detailed documentation. Your app is very good, but without documentation it seems odd to use it sometimes.

Thank you!

**Comment by damionx7** [comment]
Can you see if commits are made in `List Commits` screen in Zettel Notes ?

**Comment by jzeneto** [comment]
Hmm there's no `List Commits` screen, so I guess you're talking about `VCS (Git)` screen in left sidebar, below `Repositories`, right?

Yes, commits are made, but only locally, in `master` (my branch). If I change from `master` to `HEAD` using top right button, there are no commits for my phone changes. If I change to `origin/master`, there are no commits too.

An important thing: as soon as the app has checked out all the files the first time, there was an error saying "`HEAD` is detached", so I changed the branch of the repository to `master`. With that, subsequent pulls worked, but no push. Then I tried to put `HEAD` as branch again, same error as before. Then I tried to put `origin/master` as repository branch. This made the app create another branch in my remote repository, called literally `origin/master`, that I had to erase.
